### ISSUE TITLE
Namada chain didn't receive the ack with Hermes relaying

### ISSUE THREAD

**Issue by yito88** [heliaxdev/hermes#17]
## Summary of Bug
When a Namada chain sent a token to another Namada chain, (if the other chain didn't send, i.e. `SendPacket` event didn't happen,) the source chain didn't receive the ack though Hermes caught the `WriteAcknowledgement` event from the destination chain.

- When I executed the CLI `tx packet-ack` manually, the source chain received the ack as expected
- When both chains sent each token, both chains received each ack

## Version
`v1.7.1-namada-beta4`

## Acceptance Criteria

____

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate milestone (priority) applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by yito88** [comment]
The cause is a difference in Tendermint's event type between Cosmos and Namada.
Cosmos(Gaia) emitted a `WriteAcknowledgement` event in `Tx` event type. Namada emitted all events in `EndBlock` and they were included in `NewBlock` event type. Hermes doesn't assume that `WriteAcknowledgement` event is included in `NewBlock`. (So, transfers from Namada to Gaia worked well.)
When the event is captured in Hermes, Hermes spawns a relaying thread and relays the packet for the ack. However, Namada's `WriteAcknowledgement` event was ignored by Hermes because it was in `NewBlock` event.

I fixed Hermes to check `WriteAcknowledgement` event in `NewBlock`, and it worked well.

**Comment by hkey0** [comment]
> Bunun nedeni, Tendermint'in Cosmos ve Namada arasındaki olay türündeki farklılıktır. Cosmos(Gaia) bir mesaj yaydı `WriteAcknowledgement`olay `Tx`etkinlik tipi. Namada tüm olayları yayınladı `EndBlock`ve onlar da dahil edildi `NewBlock`etkinlik tipi. Hermes bunu varsaymıyor `WriteAcknowledgement`etkinlik dahildir `NewBlock`. (Yani Namada'dan Gaia'ya transferler işe yaradı.) Olay Hermes'te yakalandığında, Hermes bir aktarma iş parçacığı oluşturur ve paketi onay için iletir. Ancak Namada'nın `WriteAcknowledgement`olay Hermes tarafından göz ardı edildi çünkü `NewBlock`etkinlik.
> 
> Hermes'i kontrol etmesi için düzelttim `WriteAcknowledgement`olay `NewBlock`ve iyi çalıştı. [764973d ](https://github.com/heliaxdev/hermes/commit/764973dd79233e7a709b9e709763a327ebaddfdc)

hello yito
when i want to swap tokens with xcs contract in osmosis and send them to namada, the tokens get stuck in relayer.

if I then send a regular ibc transfer, the tokens stuck in relayer are transferred to namada. is this issue related to my problem?

**Comment by yito88** [comment]
Hi @hkey0, thank you for reporting it. Actually, this issue has been resolved by using the upstream `tendermint-rs`.
Could you share your operations to reproduce the issue and the versions of Namada and Hermes?

**Comment by hkey0** [comment]
> Hi @hkey0, thank you for reporting it. Actually, this issue has been resolved by using the upstream `tendermint-rs`. Could you share your operations to reproduce the issue and the versions of Namada and Hermes?

I didn't expect you to respond so quickly. So I was resting. Sorry for the late reply.


I am using the latest version of Namada (0.31.6). I installed Hermes following the instructions here. 
https://docs.namada.net/operators/ibc#from-source

I set up relayer, deployed xcs and swap router contract in osmosis testnet, set pools and channels from the contract. I can start a transaction in osmosis, swap tokens and transfer tokens to namada (atomically) but most of the time the funds get stuck in relayer. Then when I do a regular ibc transfer, the stuck funds are transferred to namada.

I sent you a message on discord (yuji088) to chat in detail. My nickname is `hkey`. Can we chat on discord for instant and fast communication? And as we are in a competition process, it may not be right to share everything publicly :|
